wallet/udb/upgrades.go

```diff
@@ -145,30 +145,36 @@ const (
 	// filters.
 	blockcf2Version = 15
 
+	// perTicketVotingPreferencesVersion is the 16th version of the database.
+	// It creates a top-level ticketsagendaprefs bucket for storing voting
+	// preferences for individual tickets.
+	perTicketVotingPreferencesVersion = 16
```

Comment:
I thought about using the existing `agendaprefs` bucket and saving per-tickets preference to transaction-hash nestedbuckets using `prefKey(version, agendaID)` but decided against that because the doc for `ReadBucket.Get` says value may be returned from nested buckets if the bucket does not contain a value for that key; meaning if no default preferences are set, trying to read default preferences could return some ticket's preferences.

Also considered using the existing `agendaprefs` bucket and modifying the key to include the ticket hash (for per-ticket reads/writes) and continue using just `version, agendaID` for default preferences but I was concerned about potential key collision, so I decided against that as well.

Reply:
yes